src/lava/magma/core/run_configs.py

```diff
@@ -70,57 +70,134 @@ def select(self,
 
     In case of `PyProcessModel`s, an appropriate `PyProcessModel` is selected
-    based on the tags associated with it. Tags are set using the `@tag`
-    decorator. If no tags are set for `ProcessModel`s and no tag is selected,
+    based on the tags associated with it (the tags are set using the `@tag`
+    decorator):
+        (a) If no tags are set for `ProcessModel`s and no tag is selected,
     then the first matching `ProcessModel` will be returned.
+        (b) If there is only one ProcessModel for a Process, it will be
+        returned irrespective of the tags associated with it.
+        (c) In all other cases, we take the queried tag and return the
+        ProcessModel matching with it
+
+    Exceptions, if any, are checked in `exception_proc_model_map` dictionary
+    as key-value pairs `{Process: ProcessModel}`. These explicit
+    specifications are given precedence over the subsequent logic for
+    `ProcessModel` selection.
+
+    Parameters
+    ----------
+    custom_sync_domains: list of synchronization domains
+    select_tag: str, ProcessModels with this tag need to be selected
+    select_sub_proc_model: bool, preferentially select SubProcessModel when
+        True and return
+    exception_proc_model_map: dict, explicit dictionary of {Process:
+        ProcessModel} classes, provided as exceptions to the ProcessModel
+        selection logic. The choices made in this dict are respected over
+        any logic. For example, {Dense: PyDenseModel}. Note that this is a
+        dict mapping classnames to classnames.
     """
 
     def __init__(self,
-                 custom_sync_domains=None,
-                 select_tag=None,
-                 select_sub_proc_model=False,
-                 exception_proc_model_map=None):
+                 custom_sync_domains: ty.Optional[ty.List[SyncDomain]] = None,
+                 select_tag: ty.Optional[str] = None,
+                 select_sub_proc_model: ty.Optional[bool] = False,
+                 exception_proc_model_map: ty.Optional[ty.Dict[
+                     ty.Type[AbstractProcess], ty.Type[
+                         AbstractProcessModel]]] = None):
         super().__init__(custom_sync_domains=custom_sync_domains)
         self.select_tag = select_tag
         self.select_sub_proc_model = select_sub_proc_model
         self.exception_proc_model_map = exception_proc_model_map
         if not exception_proc_model_map:
             self.exception_proc_model_map = {}
 
-    def select(self, proc, proc_models):
+    def select(self,
+               proc: AbstractProcess,
+               proc_models: ty.List[ty.Type[AbstractProcessModel]]) \
+            -> ty.Type[AbstractProcessModel]:
+        """
+        Selects an appropriate ProcessModel from a list of ProcessModels for
+        a Process, based on user requests
+        Parameters
+        ----------
+        proc: AbstractProcess, Process for which ProcessModel is selected
```

Comment:
var: type, text? Looks strange because it's tempting to read it as a sentence. 

Proper numpy docstring format is
var: type
    Text

Reply:
Fixed.